### PR TITLE
Reduce wasted renders for column_loading.js

### PR DESCRIPTION
**Pull request by nolanlawson**
This reduces some wasted time for this component, as shown in this `ReactPerf.printWasted()` before and after:

out2

(Scenario: load the local timeline in mobile view.)

### CODE REVIEW

File: app/javascript/mastodon/features/ui/components/column_loading.js [hunk 1]
```diff
@@ -3,17 +3,25 @@ import PropTypes from 'prop-types';
 
 import Column from '../../../components/column';
 import ColumnHeader from '../../../components/column_header';
+import ImmutablePureComponent from 'react-immutable-pure-component';
```
Comment (nolanlawson) [inline app/javascript/mastodon/features/ui/components/column_loading.js]: I tried `React.PureComponent` but for some reason that resulted in more wasted renders than `ImmuablePureComponent`.

**Review by unarist — APPROVED** [review]

File: app/javascript/mastodon/features/ui/components/column_loading.js [hunk 2]
```diff
@@ -3,17 +3,25 @@ import PropTypes from 'prop-types';
...
+  static propTypes = {
+    title: PropTypes.node,
+    icon: PropTypes.string,
+  }
 
-export default ColumnLoading;
+  render() {
+    let { title, icon } = this.props;
+    title = title || '';
+    icon = icon || '';
```
Comment (ykzts) [inline app/javascript/mastodon/features/ui/components/column_loading.js]:
```javascript
const { title = '', icon = '' } = this.props;
```

Is it not acceptable in the above writing style?

Comment (ykzts) [reply app/javascript/mastodon/features/ui/components/column_loading.js]: Ah...

```javascript
static propTypes = {
  title: PropTypes.oneOfType(PropTypes.node, PropTypes.string),
  icon: PropTypes.string,
};

static defaultProps = {
  title: '',
  icon: '',
};
```

The above may be better.

Comment (nolanlawson) [reply app/javascript/mastodon/features/ui/components/column_loading.js]: Done! :)

File: app/javascript/mastodon/features/ui/components/column_loading.js [hunk 3]
```diff
@@ -3,17 +3,28 @@ import PropTypes from 'prop-types';
...
+  };
 
-export default ColumnLoading;
+  static defaultProps = {
+    title: '',
+    icon: '',
+  };
+
+  render() {
+    let { title, icon } = this.props;
```
Comment (ykzts) [inline app/javascript/mastodon/features/ui/components/column_loading.js]: `let` -> `const`

**Pull request merged**